hermes-console-vue/src/composables/use-fetched-data/useFetchedData.ts

```diff
@@ -0,0 +1,29 @@
+import { computed, ref } from 'vue';
+import type { ComputedRef, Ref } from 'vue';
+import type { ResponsePromise } from '@/utils/axios-utils';
+
+type ReturnType<RESPONSE> = {
+  data: Ref<RESPONSE | undefined>;
+  error: Ref<boolean>;
+  isLoading: ComputedRef<boolean>;
+};
+
+export function useFetchedData<RESPONSE>({
+  request,
+}: {
+  request: () => ResponsePromise<RESPONSE>;
+}): ReturnType<RESPONSE> {
+  const data = ref<RESPONSE>();
```

Comment:
```suggestion
type ReturnType<Response> = {
  data: Ref<Response | undefined>;
  error: Ref<boolean>;
  isLoading: ComputedRef<boolean>;
};

export function useFetchedData<Response>({
  request,
}: {
  request: () => ResponsePromise<Response>;
}): ReturnType<Response> {
  const data = ref<Response>();
```
I would reserve `SCREAMING_SNAKE_CASE` only for constants. For type parameters, I would suggest using either `PascalCase` – `Response` (possibly prefixed with `T`, `TResponse`), or simply single letter names (`<T>`, or `<R>` for "Response").

[Google Style Guide for TypeScript](https://google.github.io/styleguide/tsguide.html#identifiers-type-parameters) suggests similar naming convention for type params:
> Type parameters, like in `Array<T>`, may use a single upper case character (`T`) or `UpperCamelCase`.